src/averageAges.js

```diff
@@ -36,8 +42,18 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let womenToFilter = people.filter(person => person.sex === 'f');
+
+  womenToFilter = withChildren ? womenToFilter.filter(person =>
+    people.some(child => child.mother === person.name)) : womenToFilter;
+
+  const womenTotalAge = womenToFilter.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  const womenAverageAge = womenTotalAge / womenToFilter.length;
+
+  return womenAverageAge;
```

Comment:
Repeatable code move to helper function `getAvarageAge` and reuse it everywhere where you need